python/beeai_framework/errors.py

```diff
@@ -59,114 +60,48 @@ def name(self) -> str:
         return type(self).__name__
 
     def has_fatal_error(self) -> bool:
-        """
-        Check if this error or any in the chain of exceptions under __cause__ is fatal (iterative).
-        """
-        current_exception = self  # Start with the current exception
+        current_exception: BaseException | None = self
```

Comment:
is there a reason we're removing the doc strings? I don't see a reason not to leave them if they are already written

Reply:
The method names seem self-descriptive to me. In addition, the implementation is short.